app/demo-loop/demo-loop.cc

```diff
@@ -62,6 +62,85 @@ using namespace celeritas;
 
 namespace
 {
+//---------------------------------------------------------------------------//
+//! `RootStepWriterFitler` helper function.
+bool rsw_filter_match(size_type step_attr_id, size_type filter_id)
+{
+    return filter_id == MCTruthFilter::unspecified()
+           || step_attr_id == filter_id;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * `RootStepWriter` filter.
+ *
+ * Write if event ID matches (or is undefined) *and* either the track ID or
+ * parent ID matches. If no fields are specified or are set to -1, all steps
+ * are stored.
+ */
+std::function<bool(RootStepWriter::TStepData const&)>
+root_step_writer_filter(LDemoArgs args)
```

Comment:
```suggestion
root_step_writer_filter(LDemoArgs const &args)
```
and maybe rename the function to `make_root_step_writer_filter` since it's creating a filter rather than acting as a filter?